indico/modules/events/registration/fields/base.py

```diff
@@ -8,14 +8,52 @@
 from copy import deepcopy
 from decimal import Decimal
 
-from marshmallow import fields, validate
+from marshmallow import ValidationError, fields, validate, validates
 
 from indico.core.marshmallow import mm
 from indico.modules.events.registration.models.registrations import RegistrationData
 from indico.util.marshmallow import not_empty
 
 
-class BillableFieldDataSchema(mm.Schema):
+class FieldSetupSchemaBase(mm.Schema):
+    show_if_field_id = fields.Integer(required=False, load_default=None)
+    show_if_field_value = fields.String(required=False)
+
+    @validates('show_if_field_id')
+    def _check_if_field_id(self, field_id, **kwargs):
+        from indico.modules.events.registration.models.form_fields import RegistrationFormItem
+        if field_id is None:
+            return
+        field = self.context['field']
+        if field.is_manager_only or (field.parent is not None and field.parent.is_manager_only):
+            raise ValueError('Manager-only fields cannot be conditionally shown')
+        used_field_ids = set()
+        if field.id is not None:
+            if field.id == field_id:
+                raise ValueError('The field cannot conditionally depend on itself to be shown')
+            used_field_ids.add(field.id)
+        regform = self.context['regform']
+        is_same_regform = (RegistrationFormItem.query.filter_by(
+            id=field_id, registration_form_id=regform.id).scalar() is not None)
+        if not is_same_regform:
+            raise ValidationError('The field to show does not belong to the same registration form.')
+        # Avoid cycles
+        next_field_id = field_id
+        while next_field_id is not None:
+            if next_field_id in used_field_ids:
+                raise ValidationError('Field conditions may not have cycles (a -> b -> c -> a)')
+            else:
+                used_field_ids.add(next_field_id)
+            next_field = RegistrationFormItem.query.filter_by(id=next_field_id).one_or_none()
+            if not next_field:
+                # TODO: there is a field that does not exist in the chain. Should we raise an error in this case?
```

Comment:
Can this even happen? Fields are soft-deleted, so the field should never be completely missing. --> You can use `.one()` above and just let it blow up if it still happens for whatever reason.

Reply:
I should not, but I added it just in case the logic ever gets modified (meaning broken) for whatever reason. I just left it as you suggested.